src/test/java/com/ibanity/samples/ClientSample.java

```diff
@@ -1,76 +1,141 @@
     private final String fakeTppPaymentInitiationRedirectUrl = getConfiguration("tpp.payments.initiation.result.redirect.url");
 
-    public static void main(String[] args) {
-        ClientSample clientSample = new ClientSample();
+    public ClientSample(IbanityService ibanityService) {
+        customerAccessTokenSample = new CustomerAccessTokenSample(ibanityService);
+        financialInstitutionSample = new FinancialInstitutionSample(ibanityService);
+        accountInformationAccessRequestSample = new AccountInformationAccessRequestSample(ibanityService);
+        accountSample = new AccountSample(ibanityService);
+        transactionSample = new TransactionSample(ibanityService);
+        paymentInitiationRequestSample = new PaymentInitiationRequestSample(ibanityService);
+        synchronizationSample = new SynchronizationSample(ibanityService);
+    }
+
+    public static void main(String[] args) throws CertificateException, IOException {
+        String passphrase = getConfiguration(IBANITY_CLIENT_SSL_CLIENT_PRIVATE_KEY_PASSPHRASE_PROPERTY_KEY);
+        OptionalPropertiesBuilder ibanityServiceBuilder = IbanityServiceBuilder.builder()
+                .ibanityApiEndpoint(getConfiguration(IBANITY_API_ENDPOINT_PROPERTY_KEY))
+                .applicationPrivateKey(loadPrivateKey(getConfiguration(IBANITY_CLIENT_SSL_CLIENT_PRIVATE_KEY_PATH_PROPERTY_KEY), passphrase))
+                .passphrase(passphrase)
+                .applicationCertificate(loadCertificate(getConfiguration(IBANITY_CLIENT_SSL_CLIENT_CERTIFICATE_PATH_PROPERTY_KEY)))
+                .caCertificate(loadCa(getConfiguration(IBANITY_CLIENT_SSL_CA_CERTIFICATE_PATH_PROPERTY_KEY)));
+
+        if (getConfiguration(IbanityClientSecuritySignaturePropertiesKeys.IBANITY_CLIENT_SIGNATURE_CERTIFICATE_ID_PROPERTY_KEY) != null) {
+            String signaturePassphrase = getConfiguration(IBANITY_CLIENT_SIGNATURE_PRIVATE_KEY_PASSPHRASE_PROPERTY_KEY, "");
+            ibanityServiceBuilder
+                    .requestSignaturePrivateKey(loadPrivateKey(getConfiguration(IBANITY_CLIENT_SIGNATURE_PRIVATE_KEY_PATH_PROPERTY_KEY), signaturePassphrase))
+                    .requestSignaturePassphrase(signaturePassphrase)
+                    .requestSignatureCertificate(loadCertificate(getConfiguration(IBANITY_CLIENT_SIGNATURE_CERTIFICATE_PATH_PROPERTY_KEY)))
+                    .signatureCertificateId(getConfiguration(IBANITY_CLIENT_SIGNATURE_CERTIFICATE_ID_PROPERTY_KEY));
+        }
+
+        IbanityService ibanityService = ibanityServiceBuilder.build();
+
+        ClientSample clientSample = new ClientSample(ibanityService);
+
+        CustomerAccessToken customerAccessToken = clientSample.customerAccessTokenSamples();
+        List<FinancialInstitution> financialInstitutions = clientSample.financialInstitutionSamples();
+
+        clientSample.accountInformationAccessRequestSamples(customerAccessToken, financialInstitutions);
+
```

Comment:
at this point, there should be the web navigation to authorize the accounts. We should write it, maybe in a Java comment?

Reply:
Done in the console thanks to 
` SampleHelper.waitForAuthorizationWebFlow(accountInformationAccessRequest);`